### PR TITLE
feature - profile ui

### PR DESCRIPTION
**Pull request by at-cuongdoan**

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/Define/String.swift [hunk 1]
```diff
@@ -25,6 +25,7 @@ extension App {
         static let tSignUp = "Sign up"
         static let tHome = "Home"
         static let tMap = "Map"
+        static let tProfile = "Profile"
```
Comment (at-tuanmai2) [inline MyApp/Define/String.swift]: t?

File: MyApp/Library/Ext/UIImageViewExt.swift [hunk 2]
```diff
@@ -21,4 +22,9 @@ extension UIImageView {
         }
         task.resume()
     }
+
+    func setImage(with urlString: String, placeholder: UIImage = #imageLiteral(resourceName: "im_image")) {
+        guard !urlString.isEmpty, let url = URL(string: urlString) else { return }
```
Comment (at-tuanmai2) [inline MyApp/Library/Ext/UIImageViewExt.swift]: no need isEmpty

Comment (at-tuanmai2) [inline MyApp/Library/Ext/UIImageViewExt.swift]: else set imageView = placeholder?

File: MyApp/View/Detail/DetailCollectionCell/DetailCollectionCell.swift [hunk 3]
```diff
@@ -12,14 +12,15 @@ import MVVM
 final class DetailCollectionCell: UICollectionViewCell, MVVM.View {
 
     // MARK: - IBOutlets
-    @IBOutlet weak var headerImage: UIImageView!
+    @IBOutlet fileprivate weak var headerImage: UIImageView!
 
     // MARK: - Properties
     var viewModel: DetailCollectionCellViewModel? {
         didSet {
+            headerImage.image = nil
```
Comment (at-tuanmai2) [inline MyApp/View/Detail/DetailCollectionCell/DetailCollectionCell.swift]: please call updateView Func

File: MyApp/View/Detail/DetailCollectionCell/DetailCollectionCell.swift [hunk 4]
```diff
@@ -12,14 +12,15 @@ import MVVM
 final class DetailCollectionCell: UICollectionViewCell, MVVM.View {
 
     // MARK: - IBOutlets
-    @IBOutlet weak var headerImage: UIImageView!
+    @IBOutlet fileprivate weak var headerImage: UIImageView!
 
     // MARK: - Properties
     var viewModel: DetailCollectionCellViewModel? {
```
Comment (at-tuanmai2) [inline MyApp/View/Detail/DetailCollectionCell/DetailCollectionCell.swift]: add default value (I think it better)

File: MyApp/View/Home/HomeCollection/HomeCollectionCell/HomeCollectionCell.swift [hunk 5]
```diff
@@ -12,8 +12,7 @@ import MVVM
 final class HomeCollectionCell: UICollectionViewCell, MVVM.View {
 
     // MARK: - IBOutlets
-    @IBOutlet fileprivate weak var promotionView: UIView!
-    @IBOutlet fileprivate weak var thumbnailImageView: UIImageView!
+    @IBOutlet fileprivate weak var thumbnailImage: UIImageView!
```
Comment (at-tuanmai2) [inline MyApp/View/Home/HomeCollection/HomeCollectionCell/HomeCollectionCell.swift]: ImageView?

File: MyApp/View/Home/HomeCollection/HomeCollectionCell/HomeCollectionCell.swift [hunk 6]
```diff
@@ -22,8 +21,9 @@ final class HomeCollectionCell: UICollectionViewCell, MVVM.View {
     // MARK: - Properties
     var viewModel: HomeCollectionCellViewModel? {
         didSet {
+            thumbnailImage.image = nil
             guard let viewModel = viewModel else { return }
-            thumbnailImageView.imageFromUrl(urlString: viewModel.promotion.thumImage)
+            thumbnailImage.setImage(with: viewModel.promotion.thumImage)
```
Comment (at-tuanmai2) [inline MyApp/View/Home/HomeCollection/HomeCollectionCell/HomeCollectionCell.swift]: let promotion =. ... or add struct in viewModel

File: MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift [hunk 7]
```diff
@@ -11,13 +11,40 @@ import UIKit
 final class ProfileCollectionCell: UICollectionViewCell {
 
     // MARK: - IBOutlets
-    @IBOutlet fileprivate weak var postImage: UIImageView!
+    @IBOutlet fileprivate weak var thumbnailImage: UIImageView!
```
Comment (at-tuanmai2) [inline MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift]: name?

File: MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift [hunk 8]
```diff
@@ -11,13 +11,40 @@ import UIKit
...
+
+    enum Action {
+        case success
+        case failure
+    }
+
     var viewModel: ProfileCollectionCellViewModel? {
         didSet {
             guard let viewModel = viewModel else { return }
-            postImage.image = UIImage(named: viewModel.item.thumbnail)
```
Comment (at-tuanmai2) [inline MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift]: in updateView()

File: MyApp/View/Profile/ProfileViewController.swift [hunk 9]
```diff
@@ -8,13 +8,19 @@
...
 
-final class ProfileViewController: UIViewController {
+private struct Config {
+    static let heightCell: CGFloat = 237.5
+}
+
+final class ProfileViewController: BaseController {
 
     // MARK: - IBOutlets
+    @IBOutlet fileprivate weak var avatarImage: UIImageView!
```
Comment (at-tuanmai2) [inline MyApp/View/Profile/ProfileViewController.swift]: name

File: MyApp/View/Profile/ProfileViewController.swift [hunk 10]
```diff
@@ -23,97 +29,34 @@ final class ProfileViewController: UIViewController {
...
-    private func setNavigation() {
+    private func configView() {
         title = viewModel.title
-        navigationController?.navigationBar.backgroundColor = .black
-        navigationController?.navigationBar.isTranslucent = false
-        navigationController?.tabBarController?.tabBar.isTranslucent = false
-        navigationItem.backBarButtonItem = UIBarButtonItem(title: App.String.kEmpty, style: .plain, target: nil, action: nil)
-        navigationItem.rightBarButtonItem = UIBarButtonItem(image: UIImage(named: App.String.kSetup), style: .plain, target: self, action: #selector(pressSettingButton))
-        navigationController?.navigationBar.tintColor = .black
+        navigationItem.leftBarButtonItem = UIBarButtonItem(image: #imageLiteral(resourceName: "ic_list"), style: .plain, target: self, action: #selector(showSideMenu))
```
Comment (at-tuanmai2) [inline MyApp/View/Profile/ProfileViewController.swift]: defile action is better

File: MyApp/View/Profile/ProfileViewController.swift [hunk 11]
```diff
@@ -23,97 +29,34 @@ final class ProfileViewController: UIViewController {
...
-        navigationController?.pushViewController(settingVC, animated: true)
+    // MARK: - objc Private
+    @objc private func showSideMenu() {
+        sideMenuController?.showLeftView(animated: true, completionHandler: nil)
     }
 
     // MARK: - IBAction
-    @IBAction func pressEditButton(_ sender: UIButton) {
+    @IBAction func editButtonTouchUpInside(_ sender: UIButton) {
         let editVC = ProfileEditViewController()
```
Comment (at-tuanmai2) [inline MyApp/View/Profile/ProfileViewController.swift]: let vc = is ok!

File: MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift [hunk 12]
```diff
@@ -11,13 +11,40 @@ import UIKit
...
+    }
+
+    private func addGentureRemoveView() {
+        let tapGenture = UITapGestureRecognizer(target: self, action: #selector(pressRemoveView))
+        removeView.isUserInteractionEnabled = true
+        removeView.addGestureRecognizer(tapGenture)
+    }
+
+    @objc private func pressRemoveView() {
+        complete?(.success)
```
Comment (at-tuanmai2) [inline MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift]: delegate, no use closure

File: MyApp/View/Profile/ProfileViewController.swift [hunk 13]
```diff
@@ -125,30 +68,22 @@ extension ProfileViewController: UICollectionViewDataSource {
...
-extension ProfileViewController: UITableViewDataSource {
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return viewModel.numberOfItems(inSection: section, viewType: .table)
-    }
-
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard let cell = tableView.dequeueReusableCell(withIdentifier: App.String.kProfileTableCell, for: indexPath) as? ProfileTableCell else { fatalError(App.String.kErroCell) }
-        cell.viewModel = viewModel.viewModelForItem(at: indexPath, viewType: .table) as? ProfileTableCellViewModel
-        return cell
+        return CGSize(width: (profileCollectionView.frame.size.width - 10) / 2, height: Config.heightCell)
```
Comment (at-tuanmai2) [inline MyApp/View/Profile/ProfileViewController.swift]: viewModel

File: MyApp/View/SideMenu/SideMenuViewController.swift [hunk 14]
```diff
@@ -107,6 +107,8 @@ extension SideMenuViewController: UITableViewDelegate {
             typeVC = HomeViewController()
         case .map:
             typeVC = MapViewController()
+        case .profile:
+            typeVC = ProfileViewController()
```
Comment (at-tuanmai2) [inline MyApp/View/SideMenu/SideMenuViewController.swift]: typeVc = VC?

File: MyApp/View/Profile/ProfileViewController.swift (hunk 10 above)
Comment (at-cuongdoan) [reply MyApp/View/Profile/ProfileViewController.swift]: em thấy mọi người ai cũng làm vậy mà a :D, cái ni fix kiểu chi a hè?

File: MyApp/View/Profile/ProfileViewController.swift (hunk 13 above)
Comment (at-cuongdoan) [reply MyApp/View/Profile/ProfileViewController.swift]: cái này em custom UI cho cell nên em để trong VC ạ!

File: MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift [hunk 15]
```diff
@@ -7,17 +7,53 @@
 //
 
 import UIKit
+import MVVM
 
-final class ProfileCollectionCell: UICollectionViewCell {
```
Comment (at-daiho) [inline MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift]: WARNING - Delegate protocols should be class-only so they can be weakly referenced.

File: MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift [hunk 16]
```diff
@@ -7,17 +7,50 @@
...
```
Comment (at-daiho) [inline MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift]: WARNING - Delegate protocols should be class-only so they can be weakly referenced.

File: MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift [hunk 17]
```diff
@@ -11,13 +11,40 @@ import UIKit
 final class ProfileCollectionCell: UICollectionViewCell {
 
     // MARK: - IBOutlets
-    @IBOutlet fileprivate weak var postImage: UIImageView!
+    @IBOutlet fileprivate weak var thumbnailImage: UIImageView!
+    @IBOutlet weak var removeView: UIView!
```
Comment (at-daiho) [inline MyApp/View/Profile/ProfileCollectionCell/ProfileCollectionCell.swift]: WARNING - Delegate protocols should be class-only so they can be weakly referenced.

**Comment by at-daiho** [comment]
Total linter violations in pull request: 3
Serious: 0
Warnings: 3

**Pull request merged**